controller/customdomain_controller.go

```diff
@@ -265,21 +273,25 @@ func (r *CustomDomainReconciler) Reconcile(ctx context.Context, request ctrl.Req
 			customIngress.Namespace = ingressOperatorNamespace
 			customIngress.Labels = labelsForOwnedResources()
 			customIngress.Spec.Domain = ingressDomain
-			customIngress.Spec.EndpointPublishingStrategy = &operatorv1.EndpointPublishingStrategy{
-				Type: operatorv1.LoadBalancerServiceStrategyType,
-				LoadBalancer: &operatorv1.LoadBalancerStrategy{
-					Scope: operatorv1.LoadBalancerScope(ingressScope),
-					ProviderParameters: &operatorv1.ProviderLoadBalancerParameters{
-						Type: operatorv1.LoadBalancerProviderType(operatorv1.LoadBalancerServiceStrategyType),
+			if isAWS {
+				customIngress.Spec.EndpointPublishingStrategy = &operatorv1.EndpointPublishingStrategy{
+					Type: operatorv1.LoadBalancerServiceStrategyType,
+					LoadBalancer: &operatorv1.LoadBalancerStrategy{
+						Scope: operatorv1.LoadBalancerScope(ingressScope),
+						ProviderParameters: &operatorv1.ProviderLoadBalancerParameters{
+							Type: operatorv1.LoadBalancerProviderType(operatorv1.LoadBalancerServiceStrategyType),
+						},
 					},
-				},
-			}
+				}
+			} else {
+				customIngress.Spec.EndpointPublishingStrategy = &operatorv1.EndpointPublishingStrategy{
+					Type: operatorv1.LoadBalancerServiceStrategyType,
+					LoadBalancer: &operatorv1.LoadBalancerStrategy{
```

Comment:
spec.endpointPublishingStrategy.loadBalancer.providerParameters.Type will still need to have the value "operatorv1.LoadBalancerProviderType(operatorv1.LoadBalancerServiceStrategyType)" as we still want it to be "LoadBalancerServiceStrategyType" - see https://github.com/openshift/api/blob/d170fcdc0fa638b664e4f35f2daf753cb4afe36b/operator/v1/types_ingress.go#LL348C2-L348C33

I'd suggest keeping this logic the same as before, then add a check for if cloudPlatform == "GCP", then setting the  customIngress.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.Type: operatorv1.GCPLoadBalancerProvider

Reply:
Thanks for the suggestion! I updated the code, and am running some manual tests.

Reply:
Manual test on GCP cluster passed [X]

Reply:
Does the PR also work on AWS?

Reply:
Manual test on AWS cluster passed [X]

Reply:
I did not test upgrade yet.